aip/general/0216.md

```diff
@@ -123,6 +123,44 @@ message PublishBookRequest {
 - The comment for the field **should** document the resource pattern.
 - Other fields **may** be included.
 
+### Declarative-friendly state transitions
+
+If a resource's state can be modified by [declarative tooling][aip-128], it
+**should** expose a separate `desired_state` field taking the same enum type.
+
+```proto
+message Book {
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    UNPUBLISHED = 1;
+    PUBLISHING = 2;
+    PUBLISHED = 3;
+  }
+  
+  State state = 1 [(google.api.field_behaviopr = OUTPUT_ONLY)];
+  State desired_state = 2;
+}
+```
+
+- The `state` field's output **may** change to more granular
+  [active states](#common-states) to indicate progress toward the `desired_state`.
+- The `desired_state` field **must** only allow [resting state](#common-states)
+  enum values.
+- Attempting to set `desired_state` to an [active state](#common-states) **must**
+  error with `INVALID_ARGUMENT` (HTTP 400).
+- Attempting to set `desired_state` to a state that cannot be transitioned to
+  from the current state **must** error with `FAILED_PRECONDITION` (HTTP 400).
+- A [long-running][] operation **must not** complete until the `state` has
```

Comment:
missing link?

Reply:
Markdown link quirk - automatically matches the `[long-running]` citation at the bottom. I'm cargo-culting from another instance of same in this file 😄